### ISSUE TITLE
Incoherent behaviour in 'nn.CrossEntropyLoss()' for different ways to pass 'weight'

### ISSUE THREAD

**Issue by Ailon-Island** [Jittor/jittor#405]
## Describe the bug
在使用`nn.CrossEntropyLoss()`时，指定`weight`参数为`jitter.ones(num_classes, dtype='float32')`和置为`None`的运行结果不同。且指定包括全一在内的值后会带来梯度爆炸。
## Full Log
这是指定了`weight`具体值的训练曲线。
![accuracy](https://user-images.githubusercontent.com/74121851/196984829-1bd2219b-e562-4ffa-9472-114318089e41.png)
![loss](https://user-images.githubusercontent.com/74121851/196985000-7867e00f-f93b-4bdf-bbac-d93285dbc51e.png)

这是不指定`weight`值的训练曲线。
![accuracy](https://user-images.githubusercontent.com/74121851/196985175-7f1a28ac-0e35-48e6-abf3-3c812689fe22.png)
![loss](https://user-images.githubusercontent.com/74121851/196985198-def22249-a974-4b64-809f-370afc4a6070.png)

## Minimal Reproduce

数据集是CIFAR-10，简单的Conv+Linear，类似于Jittor教程的MNIST分类网络。

## Expected behavior
希望两者训练行为至少一致，最好是都不会发生梯度爆炸。

**Comment by Exusial** [comment]
We'll fix this problem in upcoming versions, thanks for your help.
